oauth/src/main/java/ch/cyberduck/core/oauth/OAuth2RequestInterceptor.java

```diff
@@ -115,8 +115,9 @@ public void process(final HttpRequest request, final HttpContext context) throws
             if(log.isInfoEnabled()) {
                 log.info(String.format("Authorizing service request with OAuth2 tokens %s", tokens));
             }
-            request.removeHeaders(HttpHeaders.AUTHORIZATION);
-            request.addHeader(new BasicHeader(HttpHeaders.AUTHORIZATION, String.format("Bearer %s", tokens.getAccessToken())));
+            // TODO if already have the STS credentials, this adds an authorization header causing permission denied (sending OAuth credentials to STS)
```

Comment:
I see no reason there should be a OAuth handler registered for the connection to the STS service.

Reply:
You are right, we are still working on how to use the requestInterceptor only for keylcoak and not for STS